components/style/rule_tree/mod.rs

```diff
@@ -580,13 +581,13 @@ impl StrongRuleNode {
                 let existing =
                     next_sibling_ptr.compare_and_swap(ptr::null_mut(),
                                                       new_ptr,
-                                                      Ordering::SeqCst);
+                                                      Ordering::Relaxed);
```

Comment:
Hm, at first glance it would seem that we either need to manage next_sibling_ptr (and probably first_child_ptr) with acquire/release semantics, or put a barrier between the initialization of the new RuleNode and CAS-ing it into the rule tree. Otherwise, what stops the compiler from reordering the initializing writes in RuleNode::new to occur after the CAS?